### PR TITLE
[WFCORE-3561] test append file with absolute path fails on windows

### PR DESCRIPTION
**Pull request by kanovotn**

### CODE REVIEW

**Comment by jmesnil** [comment]
retest this please

**Comment by jmesnil** [comment]
This is ok to test.

**Comment by kabir** [comment]
This is ok to test

**Comment by kabir** [comment]
@jfdenise please approve

File: testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java [hunk 1]
```diff
@@ -479,29 +482,41 @@ public void testAppendCustomFileRelativeDirCompletion() throws Exception {
...
+
+            // Split the absolute path. First item must be the root directory (for X:\ on Windows)
+            paths.add(tempFile.toAbsolutePath().getRoot().toString());
+            for (int i=0; i < tempFile.toAbsolutePath().getNameCount(); i++) {
+                paths.add(tempFile.toAbsolutePath().getName(i).toString());
             }
 
             String cmd = "version >>";
             String testPath = "";
             List<String> candidates = new ArrayList<>();
```
Comment (jfdenise) [inline testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java]: No need to create an ArrayList.

File: testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java [hunk 2]
```diff
@@ -479,29 +482,41 @@ public void testAppendCustomFileRelativeDirCompletion() throws Exception {
...
-            for (int i = 0; i < paths.size(); i++) {
+            // Iterate through whole path except the last element which is the filename itself
+            for (int i = 0; i < (paths.size() - 1); i++) {
```
Comment (jfdenise) [inline testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java]: You could remove the parenthesis

File: testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java [hunk 3]
```diff
@@ -479,29 +482,41 @@ public void testAppendCustomFileRelativeDirCompletion() throws Exception {
...
-                    assertTrue(candidates.toString(), candidates.contains(p + File.separator));
+                if (i == 0 || (i == paths.size() - 1)) {
+                    testPath += p;
+                } else {
+                    testPath += p + File.separator;
+                }
+
+                // Candidate should be directory
+                if (i < tempFile.toAbsolutePath().getNameCount() - 1) {
+                    candidates = complete(ctx, cmd + testPath, null);
```
Comment (jfdenise) [inline testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java]: You could extract this line and put it prior to the branches. It is common on the 2 branches.

**Comment by kanovotn** [comment]
@jfdenise Thank you for review. I made requested changes.

**Comment by jmesnil** [comment]
retest this please

**Pull request merged**